lib/gceimgutils/gceutils.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2021 SUSE LLC
+#
+# This file is part of gceimgutils
+#
+# gceimgutils is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# gceimgutils is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with gceimgutils.ase.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import re
+
+from google.oauth2 import service_account
+from googleapiclient import discovery
+
+from gceimgutils.gceimgutilsExceptions import (
+    GCEProjectCredentialsException
+)
+
+from googleapiclient.errors import HttpError
+
+compute_api = None
+
+
+# ----------------------------------------------------------------------------
+def find_images_by_name(images, image_name, log_callback):
+    """Return a list of images that match the given name."""
+    matching_images = []
+    for image in images:
+        if not image.get('name'):
+            _no_name_warning(image, log_callback)
+            continue
+        if image_name == image['name']:
+            matching_images.append(image)
+
+    return matching_images
+
+
+# ----------------------------------------------------------------------------
+def find_images_by_name_fragment(images, image_name_fragment, log_callback):
+    """Return a list of images that match the given fragment in any part
+       of the image name."""
+    matching_images = []
+    for image in images:
+        if not image.get('name'):
+            _no_name_warning(image, log_callback)
+            continue
+        if image['name'].find(image_name_fragment) != -1:
+            matching_images.append(image)
+
+    return matching_images
+
+
+# ----------------------------------------------------------------------------
+def find_images_by_name_regex_match(images, image_name_regex, log_callback):
+    """Return a list of images that match the given regular expression in
+       their name."""
+    matching_images = []
+    image_name_exp = re.compile(image_name_regex)
+    for image in images:
+        if not image.get('name'):
+            _no_name_warning(image, log_callback)
+            continue
+        if image_name_exp.match(image['name']):
+            matching_images.append(image)
+
+    return matching_images
```

Comment:
Why not combine the 3 methods together in `find_image` and have `--image-name` flag ?

`find_images_by_name` is a subset case of `find`, so `find_images_by_name_fragment` could cover the 
case of a whole match. Combining them would reduce code duplication and iterations in the `images` loop.

Consider use `in` operator instead of find, so 
```python
image_name = 'SLES12`
image_name in 'SLES12SP4' =>True
```